sql/src/main/java/io/crate/execution/dsl/projection/builder/InputColumns.java

```diff
@@ -181,7 +181,19 @@ public Symbol visitWindowFunction(WindowFunction windowFunction, SourceSymbols s
             return replacement;
         }
         ArrayList<Symbol> replacedFunctionArgs = getProcessedArgs(windowFunction.arguments(), sourceSymbols);
-        return new WindowFunction(windowFunction.info(), replacedFunctionArgs, windowFunction.windowDefinition());
+        Symbol filterWithReplacedArgs;
+        Symbol filter = windowFunction.filter();
+        if (filter != null) {
+            filterWithReplacedArgs = filter.accept(this, sourceSymbols);
+        } else {
+            filterWithReplacedArgs = null;
```

Comment:
initial var value could be assigned to`null`

Reply:
as discussed, would keep this pattern for now